bridge/tfchain_bridge/pkg/bridge/mint.go

```diff
@@ -46,37 +61,35 @@ func (bridge *Bridge) mint(ctx context.Context, senders map[string]*big.Int, tx
 	}
 
 	if tx.Memo == "" {
-		log.Info().Str("tx_id", tx.Hash).Msg("transaction has empty memo, refunding now")
-		return bridge.refund(context.Background(), receiver, depositedAmount.Int64(), tx)
+		ctx = context.WithValue(ctx, "refund_reason", "no memo in transaction")
```

Comment:
this is not a nice way to set (and get) context. typoes and naming conflicts are a problem instead it's ususally done like this

```go
type refundReasonKey struct {}
func WithRefundReason(ctx, reason string) Context {
    return context.WithValue(ctx, refundReasonKey{}, reason)
}

func GetRefundReason(ctx) string {
    return context.GetValue(ctx, refundReasonKey{}).(string)
}
```

This makes sure
- no key conflict because a key is a type, not a value (actually it's the zero value of a type but 2 zero values of 2 different types don't equal each other)
- Clear interface to set, get a reason and reason type is clear
- no possible typoes or mistakes